### PR TITLE
Home screen

### PR DESCRIPTION
**Pull request by pashacbk**
Model is hardcoded for demo!

### CODE REVIEW

File: MeetUp/app/src/main/java/com/endava/meetup/screens/home/RVAdapter.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    public CardViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int viewType) {
+
+        View v = LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.card_view, viewGroup, false);
+        CardViewHolder cardViewHolder = new CardViewHolder(v);
+        return cardViewHolder;
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull CardViewHolder holder, int position) {
+        holder.eventImage.setImageResource(cardList.get(position).eventImageId);
```
Comment (UncuSergiu) [inline MeetUp/app/src/main/java/com/endava/meetup/screens/home/RVAdapter.java]: holder.bind(cardlist.get(position))

File: MeetUp/app/src/main/java/com/endava/meetup/screens/home/HomeActivity.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        MenuItem searchItem = menu.findItem(R.id.search_view_action);
+        SearchView searchView = (SearchView) searchItem.getActionView();
+        // Configure the search info and add any event listeners...
+
+        return super.onCreateOptionsMenu(menu);
+    }
+
+    @Override
+    public void updateUi(List<CardViewModel> data) {
+        for (int i = 0; i < 5; i++) {
```
Comment (UncuSergiu) [inline MeetUp/app/src/main/java/com/endava/meetup/screens/home/HomeActivity.java]: this for, I think should be in presenter, there just add

File: MeetUp/app/src/main/java/com/endava/meetup/screens/home/CustomRV.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+import android.support.v7.widget.LinearSnapHelper;
+import android.support.v7.widget.RecyclerView;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+
+import com.endava.meetup.R;
+
+import java.util.List;
+
+public class CustomRV extends LinearLayout {
```
Comment (UncuSergiu) [inline MeetUp/app/src/main/java/com/endava/meetup/screens/home/CustomRV.java]: for start it's ok, there is an example witch you will help you https://github.com/delaroy/Heterogeneous

File: MeetUp/app/src/main/java/com/endava/meetup/screens/auth/modules/intro/IntroFragment.java [hunk 4]
```diff
@@ -38,7 +40,8 @@ public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
         signInButton.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                presenter.onSignInClick();
+//                presenter.onSignInClick();
+                getActivity().startActivity(new Intent(getActivity(),HomeActivity.class));
```
Comment (CebotariM) [inline MeetUp/app/src/main/java/com/endava/meetup/screens/auth/modules/intro/IntroFragment.java]: please use HomeActivity.newIntent()

File: MeetUp/app/src/main/java/com/endava/meetup/screens/home/CustomRV.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+    String groupName;
+    private RecyclerView recyclerView;
+    private RecyclerView.LayoutManager layoutManager;
+    private RVAdapter rvAdapter;
+    private LinearSnapHelper snapHelper;
+    TextView groupTextName;
+
+    public CustomRV(String groupName, List<CardViewModel> cardList, Context context) {
```
Comment (CebotariM) [inline MeetUp/app/src/main/java/com/endava/meetup/screens/home/CustomRV.java]: you don't need Context in a custom view, you can call getContext()

File: MeetUp/app/src/main/res/layout/activity_home.xml [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+        android:layout_height="wrap_content"
+        app:tabTextColor="@color/primaryTextColor"
+        app:tabSelectedTextColor="@color/primaryTextColor"
+        app:tabMode="fixed">
+
+        <android.support.design.widget.TabItem
+            android:id="@+id/item1_tabLayout"
+            android:layout_width="match_parent"
+            android:layout_height="match_parent"
+            android:text="@string/text_item1_tablatout" />
```
Comment (CebotariM) [inline MeetUp/app/src/main/res/layout/activity_home.xml]: please format text - ctrl alt L

File: MeetUp/app/src/main/res/values/dimens.xml [hunk 7]
```diff
@@ -36,5 +36,9 @@
     <dimen name="textInputLayout_margin_top_bottom">8dp</dimen>
     <dimen name="spacing_150_dp">150dp</dimen>
     <dimen name="signin_email_margintop">190dp</dimen>
+    <dimen name="spacing_170_dp">170dp</dimen>
+    <dimen name="spacing_180_dp">180dp</dimen>
```
Comment (CebotariM) [inline MeetUp/app/src/main/res/values/dimens.xml]: you won't reuse these values, please name them for the specific scenario you are using

File: MeetUp/app/src/main/res/values/strings.xml [hunk 8]
```diff
@@ -18,4 +18,6 @@
     <string name="signUp_pass_hint_text">Password</string>
     <string name="signUp_login_btt">Login</string>
     <string name="signUp_email_hint_text">Email</string>
+    <string name="search_action">Search</string>
+    <string name="text_item1_tablatout">All events</string>
```
Comment (CebotariM) [inline MeetUp/app/src/main/res/values/strings.xml]: avoid using names like text_item1

File: MeetUp/app/src/main/java/com/endava/meetup/screens/home/CustomRV.java [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+import java.util.List;
+
+public class CustomRV extends LinearLayout {
+
+    String groupName;
+    private RecyclerView recyclerView;
+    private RecyclerView.LayoutManager layoutManager;
+    private RVAdapter rvAdapter;
+    private LinearSnapHelper snapHelper;
+    TextView groupTextName;
```
[the remark this answers is not in the document]
Comment (CebotariM) [reply MeetUp/app/src/main/java/com/endava/meetup/screens/home/CustomRV.java]: why did you remove the private, do you need to use them somewhere else? encapsulation?

File: MeetUp/app/src/main/res/values/dimens.xml [hunk 10]
```diff
@@ -36,9 +36,7 @@
     <dimen name="textInputLayout_margin_top_bottom">8dp</dimen>
     <dimen name="spacing_150_dp">150dp</dimen>
     <dimen name="signin_email_margintop">190dp</dimen>
-    <dimen name="spacing_170_dp">170dp</dimen>
-    <dimen name="spacing_180_dp">180dp</dimen>
+    <dimen name="card_height">180dp</dimen>
     <dimen name="spacing_250_dp">250dp</dimen>
```
Comment (CebotariM) [inline MeetUp/app/src/main/res/values/dimens.xml]: you won't need spacing 250_dp either, rename it to the scenario you are using

File: MeetUp/app/src/main/java/com/endava/meetup/screens/home/CardViewHolder.java [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+import android.support.v7.widget.RecyclerView;
+import android.view.View;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import com.endava.meetup.R;
+
+public class CardViewHolder extends RecyclerView.ViewHolder {
+
+    CardView cardView;
```
Comment (UncuSergiu) [inline MeetUp/app/src/main/java/com/endava/meetup/screens/home/CardViewHolder.java]: encapsulation ?  add private for all variables

File: MeetUp/app/src/main/java/com/endava/meetup/screens/home/CustomRV.java [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+import android.widget.LinearLayout;
+import android.widget.TextView;
+
+import com.endava.meetup.R;
+
+import java.util.List;
+
+public class CustomRV extends LinearLayout {
+
+    String groupName;
```
Comment (UncuSergiu) [inline MeetUp/app/src/main/java/com/endava/meetup/screens/home/CustomRV.java]: encapsulation ? add private for all variables

**Pull request merged**